tsconfig.json

```diff
@@ -28,4 +28,10 @@
     "include": [
         "pkg/**/*"
     ],
+    "exclude": [
+        // these are part of the build system, not the app
```

Comment:
Yes, they are not part of the app, but they are still JavaScript, and checking types there is just as valid as checking our React code. This just smells like a workaround for https://github.com/glromeo/esbuild-sass-plugin/issues/183, but for my taste this is too big a hammer.

Reply:
This is conceptually wrong, though.  These are effectively part of a different program, running in a different context.

For example, since we list `"lib": "dom"` in our `tsconfig`, it means that these files could access global DOM-related functions (which they're incapable of, of course).

If we did intend to run meaningful typechecking on this code then we'd want to setup a second tsconfig (possibly via `tsc --build`) and explicitly include them as part of it.  In that case, we absolutely would want to have `@types/node` present, as it gets used from that code.  I consider it a bit of a moot point, though, since a) we disable warnings for JS, b) none of this stuff is in TS, and c) none of this stuff could easily be in TS, since `node` doesn't load `.ts` files.  There are third party plugins for that, but we don't use them...

So ya, we've been driven to clean this up because of the bug in the sass plugin, but conceptually, this is definitely the right thing to do.  If you like, I can investigate a multi-`tsconfig` setup and `tsc --build` in order to also check our build system code, but it seems like an extremely pointless theoretical exercise, considering the points above.

Reply:
Thanks for the explanation! Nah, makes sense, let's not do the multi-tsconfig setup.